zipline/algorithm.py

```diff
@@ -383,6 +384,40 @@ def __init__(self, *args, **kwargs):
 
         self.benchmark_sid = kwargs.pop('benchmark_sid', None)
 
+        self.capital_changes = kwargs.pop('capital_changes', {})
+        if self.capital_changes:
+            self.capital_changes = \
```

Comment:
My main suggestion was that they be passed from quantopian's frontend to the executor as a string(ISO-8601 timestamp) because that's a format that can be easily sent over the wire. It could be that zipline works directly with the appropriate date object, and we take care of parsing strings to dates upstream